src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,194 @@
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when age is null");
+  }
+
+  @Test
+  void shouldThrowException_whenAgeIsNegative() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword("validPassword123");
+    user.setAge(-25);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when age is negative");
+  }
+
+  @Test
+  void shouldThrowException_whenAgeIsBelowMinimum() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword("validPassword123");
+    user.setAge(17);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when age is below minimum required");
+  }
+
+  @Test
+  void shouldThrowException_whenLoginAlreadyExists() {
+    User user = new User();
+    user.setLogin("validLogin");
+    user.setPassword("validPassword123");
+    user.setAge(50);
+    Storage.people.add(user);
+    assertThrows(UserDataInvalidException.class, () -> {
+              registrationService.register(user);
+            },
+            "Expected UserDataInvalidException when login already exists in storage");
+  }
+
+  @Test
+  void shouldRegisterValidUser() {
+    User expected = new User();
+    expected.setLogin("validLogin");
+    expected.setPassword("validPassword123");
+    expected.setAge(50);
+    User actual = registrationService.register(expected);
+    assertEquals(expected, actual, "Expected the registered user to match the input user data");
+  }
+
+  @Test
+  void shouldRegisterUser_whenAllFieldsAreAtEdgeCase() {
+    User expected = new User();
+    expected.setLogin("login6");
+    expected.setPassword("123456");
+    expected.setAge(18);
+    User actual = registrationService.register(expected);
+    assertEquals(expected, actual, "Expected the registered user to match the input user data for edge case");
+  }
+
+  @Test
+  void shouldAddUserToStorage_whenRegistrationIsSuccessful() {
+    User expected = new User();
+    expected.setLogin("validLogin");
+    expected.setPassword("validPassword123");
+    expected.setAge(50);
+    User actual = registrationService.register(expected);
+    assertTrue(Storage.people.contains(actual), "Expected the storage to contain the registered user");
+  }
+
+  @Test
+  void shouldNotAddUserToStorage_whenRegistrationFails() {
+    User user = new User();
+    user.setLogin(null);
+    user.setPassword("validPassword123");
+    user.setAge(50);
+    assertThrows(UserDataInvalidException.class, () -> {
+      registrationService.register(user);
+    }, "Expected UserDataInvalidException for user with null login");
+    assertFalse(Storage.people.contains(user), "Expected the storage not to contain the user when registration fails");
```

Comment:
The test cases should be independent and not rely on the other tests to fill the data. It's good you clear the storage before each test, but ensure that each test case sets up its own necessary data without assuming the state left by a previous test.